src/ui/Interface.js

```diff
@@ -25,10 +26,25 @@ module.exports = function Interface (options) {
     function changeVal(result) {
 
       if($("#"+options.selector).val())
-        $("#location").val($("#"+options.selector).val());
+        $("#"+options.locationText).val($("#"+options.selector).val());
 
-      else if(result.results[0])
-        $("#location").val(result.results[0].formatted_address);
+      else if(result.results[0]) {
+        if(options.getPrecision() <=0 ) {
```

Comment:
Hmm, here, what if we simply made the request to `getPlacenameFromCoordinates()` for a specific "resolution" -- like, 

```js
getPlacenameFromCoordinates(options.getLat(), options.getLon(), options.getPrecision(), onPlacenameReturned)
```

and we deal with the different types in the `getPlacenameFromCoordinates()` definition? That way, if people wanted to connect up a different geocoder, and the response were differently formatted, we wouldn't have to worry about it in the interface code over here. `getPlacenameFromCoordinates()` could return a string, no more.